### PR TITLE
cicd: add git related environment variables in later image layer to a…

### PR DESCRIPTION
**Pull request by geekflyer**
…void rebuilding the rust code every time

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2512)

### CODE REVIEW

File: docker/rust-all.Dockerfile [hunk 1]
```diff
@@ -67,6 +58,14 @@ EXPOSE 6186
 ENV RUST_BACKTRACE 1
 ENV RUST_LOG_FORMAT=json
 
+# add build info
```
Comment (rustielin) [inline docker/rust-all.Dockerfile:61]: why do we need this in `builder`?

**Review by rustielin — APPROVED** [review]

File: docker/rust-all.Dockerfile (hunk 1 above)
Comment (geekflyer) [reply docker/rust-all.Dockerfile:61]: good catch, we don't

Comment (geekflyer) [reply docker/rust-all.Dockerfile:61]: actually it's not in builder :)

File: docker/rust-all.Dockerfile [hunk 2]
```diff
@@ -67,6 +58,14 @@ EXPOSE 6186
...
+ARG GIT_TAG
+ENV GIT_TAG ${GIT_TAG}
+ARG GIT_BRANCH
+ENV GIT_BRANCH ${GIT_BRANCH}
+ARG GIT_SHA
+ENV GIT_SHA ${GIT_SHA}
```
Comment (ibalajiarun) [inline docker/rust-all.Dockerfile:67]: @geekflyer we are using `option_env` macros that will be evaluated only at build time. We should change that as well, if we want to pursue this way.

**Pull request merged**